Tests/BotTests/Interceptor/Coordinator.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+
+private final class RecordingContext {
+    var destination: URL
+    var recordedStubs: [Interceptor.Stub] = []
+    var session: URLSession
+
+    init(destination: URL) {
+        self.destination = destination
+        self.session = URLSession(configuration: {
+            let config = URLSessionConfiguration.default
+            config.requestCachePolicy = .reloadIgnoringLocalCacheData
+            config.urlCache = nil
+            return config
+        }())
+    }
+}
+
+final class Coordinator {
+    private var stubs: [Interceptor.Stub] = []
+    private var recordingContext: RecordingContext? = nil
+
+    var isRecordingEnabled: Bool {
+        return recordingContext != nil
+    }
+
+    deinit {
+        Interceptor.stopRecording()
+    }
+
+    func load(stubs newStubs: [Interceptor.Stub]) {
+        stubs.append(contentsOf: newStubs)
+    }
+
+    func loadOrRecordStubs(into url: URL) {
+
+        if FileManager.default.fileExists(atPath: url.absoluteString) {
```

Comment:
So does that mean that in order to re-record new stubs, there's no flag to turn on/off (like `self.recordMode = true`) in the code? It will just be in replay mode if file is present, and record mode if file is absent, so to re-record you just delete the file?

That's cool but does that also mean that if we forget to commit the recorded file, and thus the CI won't have it, it will be in record mode (thus hitting the real API) every time the CI runs, without us noticing that we forgot to commit? Or is there an automatic mechanism that makes the tests fail if we're in `recordMode`, to be sure CI is never in record mode accidentally but record mode is only used locally while working on a feature?

Reply:
Very good point 👍 I'll include a flag to ensure this gets disabled by default leading to a failure over hitting the real API.